yaksh/code_evaluator.py

```diff
@@ -108,32 +108,41 @@ def safe_evaluate(self, user_answer, test_case_data, file_paths=None):
         # Add a new signal handler for the execution of this code.
         prev_handler = create_signal_handler()
         success = False
+        error = ""
+        weightage = 0
 
         # Do whatever testing needed.
         try:
             for test_case in test_case_data:
                 success = False
                 self.compile_code(user_answer, file_paths, **test_case)
-                success, err = self.check_code(user_answer, file_paths, **test_case)
-                if not success:
-                    break
+                success, err, test_case_weightage = self.check_code(user_answer,
+                    file_paths,
+                    partial_grading,
+                    **test_case
+                )
+                if success:
+                    weightage += test_case_weightage
+                    error = err
+                else:
+                    error += err + "\n"
```

Comment:
So for all testcases we are appending the err.
In case, the first testcase passes and the next testcase fails then
**error message will be Correct Answer + Assertion Error.**

This can be covered in your next PR related to error message formatting,